File: backend/src/main/java/com/eventflow/eventflow_backend/entity/Event.java
package com.eventflow.eventflow_backend.entity;

import com.eventflow.eventflow_backend.entity.enums.EventStatus;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.OffsetDateTime;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "events")
@Getter
@Setter
@NoArgsConstructor
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "creator_id", nullable = false)
    private User creator;

    @Column(name = "title", nullable = false, length = 255)
    private String title;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "starts_at", nullable = false)
    private OffsetDateTime startsAt;

    @Column(name = "address", nullable = false, length = 500)
    private String address;

    @Column(name = "lat")
    private Double lat;

    @Column(name = "lon")
    private Double lon;

    @Column(name = "poster_path", length = 500)
    private String posterPath;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 32)
    private EventStatus status;

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;
}

File: frontend/src/types/event.ts
export type EventStatus = "ACTIVE" | "CANCELLED";

export interface EventRequest {
  title: string;
  description?: string | null;
  startsAt: string;
  address: string;
  lat?: number | null;
  lon?: number | null;
}

export interface EventResponse {
  id: number;
  creatorId: number;
  title: string;
  description: string | null;
  startsAt: string;
  address: string;
  lat: number | null;
  lon: number | null;
  posterPath: string | null;
  posterUrl: string | null;
  status: EventStatus;
  createdAt: string;
  updatedAt: string;
}

export interface FileUploadResponse {
  objectName: string;
  objectPath: string;
  objectUrl: string;
  bucket: string;
}